typescript/src/schema/proto.ts

```diff
@@ -442,32 +448,28 @@ function augmentService(
       '.google.api.oauthScopes'
     ].split(',');
   }
-  augmentedService.pathTemplate = [];
+  augmentedService.pathTemplates = [];
   for (const property of Object.keys(messages)) {
     const m = messages[property];
-    if (m && m.options) {
-      const option = m.options;
-      if (option && option['.google.api.resource']) {
-        const opt = option['.google.api.resource'];
-        const onePathTemplate = option[
-          '.google.api.resource'
-        ] as ResourceDescriptor;
-        if (opt.type) {
-          const arr = opt.type.match(/\/([^.]+)$/);
-          if (arr) {
-            opt.type = arr[arr.length - 1];
-          }
-        }
-        const pattern = opt.pattern;
-        //TODO: SUPPORT MULTIPLE PATTERNS
-        if (pattern && pattern[0]) {
-          const params = pattern[0].match(/{[a-zA-Z]+}/g) || [];
-          for (let i = 0; i < params.length; i++) {
-            params[i] = params[i].replace('{', '').replace('}', '');
+    if (m && m.field) {
+      const fields = m.field;
+      for (const fieldDescriptor of fields) {
+        if (fieldDescriptor && fieldDescriptor.options) {
+          const option = fieldDescriptor.options;
+          if (option && option['.google.api.resourceReference']) {
+            const resourceReference = option['.google.api.resourceReference'];
+            const type = resourceReference.type;
+            if (!type || !resourceMap[type.toString()]) {
+              console.warn(
+                'resource does not have a type or resource is not in the map'
```

Comment:
1. Please capitalize the first letter :)

2. This warning does not give any actionable information: what file is it, which resource is bad, what map is it (the user has no idea about our map). The warning messages should be actionable. E.g. `In FILE.proto, message MESSAGE refers to an unknown resource RESOURCE`, or something like this.

Reply:
Sure! Added useful warning messages